examples/rest-api/main.py

```diff
@@ -0,0 +1,354 @@
+import os
+import yaml
+from fastapi import FastAPI, UploadFile, Depends, HTTPException
+from sqlalchemy.orm import Session
+from embedchain import Pipeline as App
+from embedchain.client import Client
+from models import (
+    QueryApp,
+    SourceApp,
+    MessageApp,
+    DefaultResponse,
+    DeployAppRequest,
+    # SetEnvKeys,
+)
+from database import Base, engine, SessionLocal
+from services import get_app, save_app, get_apps, remove_app
+
+Base.metadata.create_all(bind=engine)
+
+
+def get_db():
+    db = SessionLocal()
+    try:
+        yield db
+    finally:
+        db.close()
+
+
+app = FastAPI(
+    title="Embedchain REST API",
+    description="This is the REST API for Embedchain.",
+    version="0.0.1",
+    license_info={
+        "name": "Apache 2.0",
+        "url": "https://github.com/embedchain/embedchain/blob/main/LICENSE",
+    },
+)
+
+
+@app.get("/ping", tags=["Utility"])
+def check_status():
+    """
+    Endpoint to check the status of the API.
+    """
+    return {"ping": "pong"}
+
+
+# FIXME: Put on hold for now
+# @app.put("/env", tags=["Apps"])
+# async def set_env(body: SetEnvKeys):
+#     """
+#     Set the Environment variables needed for respective LLMs to work.\n
+#     """
+#     keys = body.keys
+#     try:
+#         for key in keys:
+#             os.environ[key] = keys[key]
+
+#         return DefaultResponse(response="Env variables set successfully.")
+#     except Exception as e:
+#         raise HTTPException(detail=f"Error setting API key: {e}", status_code=400)
+
+
+@app.get("/apps", tags=["Apps"])
+async def get_all_apps(db: Session = Depends(get_db)):
+    """
+    Get all apps.
+    """
+    apps = get_apps(db)
+    return {"data": apps}
+
+
+@app.post("/create", tags=["Apps"], response_model=DefaultResponse)
+async def create_app_using_default_config(app_id: str, db: Session = Depends(get_db)):
+    """
+    Create a new app using App ID.
+    This uses a default config
+    i.e, our opensource config that doesn't require any API key to run.\n
+    """
+    try:
+        if app_id is None:
+            raise HTTPException(detail="App ID not provided.", status_code=400)
+
+        if get_app(db, app_id) is not None:
+            raise HTTPException(detail=f"App with id '{app_id}' already exists.", status_code=400)
+
+        yaml_path = "default.yaml"
+
+        save_app(db, app_id, yaml_path)
+
+        return DefaultResponse(response=f"App created successfully. App ID: {app_id}")
+    except Exception as e:
+        raise HTTPException(detail=f"Error creating app: {e}", status_code=400)
+
+
+@app.post("/create/yaml", tags=["Apps"], response_model=DefaultResponse)
+async def create_app_using_custom_config(config: UploadFile, db: Session = Depends(get_db)):
```

Comment:
Both API /create and /create/yaml should be one API only.  If user doesn't pass any config yaml then we use default one.

Reply:
Got it!